integration_test/third_party_apps_test.go

```diff
@@ -376,9 +375,17 @@ func runSingleTest(ctx context.Context, logger *logging.DirectoryLogger, vm *gce
 		if err != nil {
 			return nonRetryable, fmt.Errorf("could not unmarshal contents of metadata.yaml: %v", err)
 		}
+		if err = validate.Struct(&metadata); err != nil {
+			return nonRetryable, fmt.Errorf("could not validate contents of metadata.yaml: %v", err)
+		}
 		logger.ToMainLog().Printf("Parsed metadata.yaml: %+v", metadata)
 	}
 
+	if _, err = runScriptFromScriptsDir(
+		ctx, logger, vm, path.Join("applications", app, folder, "install"), nil); err != nil {
+		return retryable, fmt.Errorf("error installing %s: %v", app, err)
+	}
+
```

Comment:
Reorder this after yaml parsing and validation as there is no point on starting the VMs if the metadata.yaml was wrong in the first place